include/ideep/operators/inner_product.hpp

```diff
@@ -333,20 +349,29 @@ struct inner_product_backward_weights
         : primitive_desc({src_desc, diff_weights_desc, diff_dst_desc},
                           aengine, forward_hints);
 
+    // reorder diff_dst(grad_y), diff_weights(grad_w), src if needed
     auto expected_diff_dst = diff_dst.reorder_if_differ_in(pd.diff_dst_desc());
     auto expected_src = src.reorder_if_differ_in(pd.src_desc());
-    diff_weights.reinit_if_possible(pd.diff_weights_desc());
+    if (diff_weights.is_empty())
+      diff_weights.reinit_if_possible(pd.diff_weights_desc());
+    auto expected_diff_weights = diff_weights.reorder_if_differ_in(pd.diff_weights_desc());
 
     exec_args args {{DNNL_ARG_DIFF_DST, expected_diff_dst},
                     {DNNL_ARG_SRC, expected_src},
-                    {DNNL_ARG_DIFF_WEIGHTS ,diff_weights}};
+                    {DNNL_ARG_DIFF_WEIGHTS ,expected_diff_weights}};
 
+    ideep::tensor expected_diff_bias;
     if (with_diff_bias) {
-      diff_bias.reinit_if_possible(pd.diff_bias_desc());
-      args.insert({DNNL_ARG_DIFF_BIAS, diff_bias});
+      // reorder diff_bias(grad_b)
+      if (diff_bias.is_empty())
+        diff_bias.reinit_if_possible(pd.diff_bias_desc());
+      expected_diff_bias = diff_bias.reorder_if_differ_in(pd.diff_bias_desc());
+      args.insert({DNNL_ARG_DIFF_BIAS, expected_diff_bias});
```

Comment:
For bias, Do it has many format? if not, i don't think we need change those code and we don't need check diff_bias whether is empty.

Reply:
Thanks, @XiaobingSuper.
I only observer 1 format for bias. But I still wish to query this in case oneDNN have some changes.

We need to check whether diff_bias is empty because I some user cases I allocate buffer in aten, if here wo got an un-empty tensor, means we should writing to this buffer provided by userer.